### ISSUE TITLE
Dependency not refreshed in my use case 

### ISSUE THREAD

**Issue by itissid** [nebula-plugins/gradle-dependency-lock-plugin#103]
If I use `./gradlew deleteLock updateLock saveLock` the lock for my dynamic dependency is not updated.
But when executed separately `./gradlew deleteLock` and `./gradlew  updateLock saveLock` updates the dependency in my project buildDir. What am I missing?

**Comment by itissid** [comment]
It seems that if i run the tasks in same gradle runtime the plugin code is configured to hold on to a file handle to the lock file. And that can of course will cause the behavior I expect. Any way of avoiding that for my use case?

**Comment by itissid** [comment]
Actually I figured out what the exact issue is. Its the fact that when the plugin is applied using `apply plugin:nebula.dependency-lock` that the configurations are resolved and those that are in the lock file are set in place. This is still a bit counter intuitive for me.  I have worked around it by using a combination of updateDependencies and the override parameters for the updateLocka and saveLock tasks. 

I suppose that's a reasonable way to do this.
